geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/index/GeoMesaSchemaValidator.scala

```diff
@@ -20,10 +20,31 @@ object GeoMesaSchemaValidator {
   def validate(sft: SimpleFeatureType): Unit = {
     MixedGeometryCheck.validateGeometryType(sft)
     TemporalIndexCheck.validateDtgField(sft)
+    ReservedWordCheck.validateAttributeNames(sft)
   }
 }
 
 /**
+  * Utility object that ensures that none of the (local portion of the) property
+  * names is a reserved word in ECQL.  Using those reserved words in a simple
+  * feature type will cause queries to fail.
+  */
+object ReservedWordCheck extends LazyLogging {
+
+  // ensure that no attribute names are reserved words within GeoTools that will cause query problems
+  def validateAttributeNames(sft: SimpleFeatureType): Unit = {
+    val reservedWords = FeatureUtils.sftReservedWords(sft)
+    if (reservedWords.nonEmpty) {
+      // TODO:  Make this an exception after GeoMesa 1.2.2
```

Comment:
Ooh... @jnh5y and I discussed this one.  If we make it an exception now -- which is how it was during debugging, so that we could find all of the unit tests that broke -- then we will necessarily break code that may have been working for folks.  (For example, if you ingested with a field named "between", but never queried against it.)  We decided that being less invasive up front was warranted; we can always tighten up more easily than we can re-release to relax a constraint.  Your thoughts?